src/frontend/src/components/Map.js

```diff
@@ -143,11 +156,39 @@ const Map = ({
   hullPoints,
   predictionHexagonIDs,
   annotationHexagonIDs,
-  onAddAnnotationHexagonIDs,
   hexResolution,
   taxonId,
+  onAddAnnotationHexagonIDs,
+  onAddAnnotationMultiSelect,
 }) => {
   console.log('Render map');
+
+  const [multipleAnnotationHexagonIDs, setMultipleAnnotationHexagonIDs] = useState(null);
+  const [isAddAnnotationMultiSelect, setIsAddAnnotationMultiSelect] = useState(true);
+
+  const handleCreated = (e) => {
+    try {
+      const layer = e.layer;
+      const polygonCoords = layer.getLatLngs()[0].map((latlng) => [latlng.lat, latlng.lng]);
+      var hexagonIds = h3.polyfill(polygonCoords, hexResolution);
+      // Add hexagons for each vertex of the polygon
+      polygonCoords.map((latlng) => hexagonIds.push(h3.geoToH3(latlng[0], latlng[1], hexResolution)));
+    // catch an error if the figure is not fully drawn
+    } catch (error) {
+      var hexagonIds = null;
+    }
+      // Clean blue select layer 
+    e.layer.remove();
+    setMultipleAnnotationHexagonIDs(hexagonIds);
+    // Set to true to add hexagons if the user draws a polygon; 
+    // set to false to remove hexagons if the user draws a rectangle.
+    setIsAddAnnotationMultiSelect(e.layerType === 'polygon');
+  };
+  
+  useEffect(() => {
```

Comment:
onAddAnnotationMultiSelect is missing in useeffect dependency, I don't think it will change, but you can add it to resolve lint error